service/matching/matchingEngine.go

```diff
@@ -617,23 +617,24 @@ func (e *matchingEngineImpl) listTaskQueuePartitions(request *matchingservice.Li
 		*request.TaskQueue,
 		taskQueueType,
 	)
+
 	if err != nil {
 		return nil, err
 	}
+
 	partitionHostInfo := make([]*taskqueuepb.TaskQueuePartitionMetadata, len(partitions))
```

Comment:
i am not super-clear what this behavior means in the context of how the feature is supposed to work, but i think this fixes the panic!

Reply:
agreed. priority is to fix the panic, but i dont think this is doing anything meaningful beyond that :)